### PR TITLE
Feature: vigem driver for ddrio

### PR DESCRIPTION
**Pull request by DinsFire64**
Reads state of ddrio and updates vigem pads. Similar in vein to viggem-sdvxio and vigem-iidxio

Useful for playing xinput related games with official hardware.

### CODE REVIEW

**Comment by icex2** [comment]
Really nice one, looking forward to giving this a try on my own cabinet.

I noticed we have some brief docs for the iidxio and sdvxio vigem clients. It would be cool to have something similar to these for the ddrio one as well and link it from the doc/vigem/README.md.

File: src/main/vigem-ddrio/config-vigem-ddrio.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            CCONFIG_CMD_USAGE_OUT_STDOUT)) {
+        cconfig_finit(config);
+        return false;
+    }
+
+    vigem_ddrio_config_get(config_out, config);
+
+    cconfig_finit(config);
+
+    // TODO: check bounds of config inputs here.
```
Comment (icex2) [inline src/main/vigem-ddrio/config-vigem-ddrio.c:62]: What config inputs?

File: src/main/vigem-ddrio/main.c [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        if (config.enable_reactive_light) {
+            set_reactive_lights(pad_state);
+        }
+
+        if (check_key(pad_state, DDR_TEST) &&
+            check_key(pad_state, DDR_SERVICE)) {
+            loop = false;
+        }
+
+        // avoid banging
```
Comment (icex2) [inline src/main/vigem-ddrio/main.c:231]:
```suggestion
        // avoid CPU banging
```

:P

**Comment by icex2** [comment]
Code's looking good. Give me a ping using the "re-request review" feature and I will have a another peek and merge this asap.

File: src/main/vigem-ddrio/config-vigem-ddrio.c (hunk 1 above)
Comment (DinsFire64) [reply src/main/vigem-ddrio/config-vigem-ddrio.c]: Typo from using vigem-sdvxio as template. Fixed.

**Comment by icex2** [comment]
Great work, thanks. Merged.

**Pull request merged**